tests/integration/setup.coffee

```diff
@@ -73,6 +73,14 @@ exports.resetUser = ->
 
 			resin.pine.delete
 				resource: 'user__has__public_key'
+
+			# logins with api keys don't have
+			# access to the api key resource
+			if loggedInWithApiKey then Promise.resolve()
+			else resin.pine.delete
+				resource: 'api_key'
+				# only delete named user api keys
+				options: $filter: name: $ne: null
```

Comment:
This special case flag is kind of nasty. Easier option, remove that extra noise, and change this to just:

```coffee
resin.pine.delete
	resource: 'api_key'
    # only delete named user api keys
    options: $filter: name: $ne: null
# Api keys can't delete api keys, just ignore failures here
.catchReturn()
```

Reply:
I would prefer a way to detect whether we are using a session token or an api key but anyway :)